packages/trpc-server/src/db.ts

```diff
@@ -99,15 +99,40 @@ export type SelectPlaylistProps = {
   ownerId?: number
 }
 
+type AugmentedPlaylistRows = (APlaylist & {
+  artistHandle?: string
+})[]
+
 export async function selectPlaylistsCamel(p: SelectPlaylistProps) {
-  return sql<APlaylist[]>`
-    select ${p.cols ? sql(p.cols) : sql`*`}
-    from playlists
+  const rows = await sql<AugmentedPlaylistRows>`
+    select users.handle as artist_handle, ${
+      p.cols ? sql(p.cols) : sql`pl_with_slug.*`
+    }
+    from (
+      select *, (
+        select slug
+        from playlist_routes pr
+        where
+        pr.playlist_id = playlists.playlist_id and is_current = 'true'
+        )
+      from playlists
+    ) as pl_with_slug
+    left join users on playlist_owner_id = user_id
     left join aggregate_playlist using (playlist_id, is_album)
-    where is_current = true
+    where pl_with_slug.is_current = true
     ${p.isAlbum != undefined ? sql`and is_album = ${p.isAlbum}` : sql``}
     ${p.ids ? sql`and playlist_id in ${sql(p.ids)}` : sql``}
     ${p.ownerId ? sql`and playlist_owner_id = ${p.ownerId}` : sql``}
-    order by created_at desc
+    order by pl_with_slug.created_at desc
   `
+
+  if (p.cols && !p.cols?.includes('permalink')) return rows
```

Comment:
That's interesting so some entities have permalinks?

Reply:
no, this query is only for playlists (collections)

since you can provide columns to select in the `p` args, we can skip the permalink calculation when you didn't ask for it or all columns (default)

now that I'm looking at it though, I think we still want to call `delete row.artistHandle` on each one. Or maybe I can rewrite the sql so it doesn't select it in the first place... 🤔